### ISSUE TITLE
Lambda remaining execution time via an API?

### ISSUE THREAD

**Issue by jsumners** [aws/aws-sdk-js#4026]
Confirm by changing [ ] to [x] below:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [ ] This is an issue with version 2.x of the SDK

**Describe the question**

I understand that the [context object](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html) has a `getRemainingTimeInMillis` method. But in order to use that in a transient dependency, the user of the transient dependency must remember to pass this context object along. My goal is to write an HTTP client library that can be used in non-Lambda code as well as in a Lambda function. When used within a Lambda function, this HTTP client should be able to determine what remaining execution time is available before issuing a request so that it can manage timeouts accordingly. Given that, is there an SDK method this transient HTTP client dependency can use to get this information without having to rely on the context object?

**Comment by vudh1** [comment]
Hi @jsumners thanks for reaching out. To confirm, are you asking for a `getRemainingTimeInMillis` method that works with non-Lambda function?

**Comment by jsumners** [comment]
No. The execution environment is within a Lambda function. I just want to be able to get the desired information without requiring developers to pass the Lambda `context` object to the HTTP client.
